api/src/main/java/org/openmrs/hl7/HL7InQueueProcessor.java

```diff
@@ -28,8 +28,8 @@
 public class HL7InQueueProcessor /* implements Runnable */{
 	
 	private static final Logger log = LoggerFactory.getLogger(HL7InQueueProcessor.class);
-	
-	private static Boolean isRunning = false; // allow only one running
```

Comment:
I thought you would not touch the isRunning variable but only that you do not use it for synchronization.

Reply:
Look at what @JudeNiroshan said. But ok, I can rever that.

Reply:
@dkayiwa Fixed :)

Reply:
@dkayiwa I think leaving the isRunning variable untouched is not good. As that is a private variable which is only being used solely for synchronization purpose within this class, we are good to replace instead of introducing another new variable. (It's just my personal opinion :) )

Reply:
@JudeNiroshan if you take a second look, you will notice that using it as a flag for the running status is quite different from locking. Just that originally, it was used for both saving the running status and locking. You can use a lock for a block of code. Removing the code block and leaving only the lock is what you are suggesting. 😊  Does it make sense?

@BartlomiejRasztabiga how about you, what do you say? 😊

Reply:
@dkayiwa I totally agree with you!

Reply:
@dkayiwa "Removing the code block and leaving only the lock is what you are suggesting."
Yes, exactly! I don't understand the requirement of both having a lock and boolean flag together. I think the requirement is to run only one execution in JVM for a given time period. If that so, having a lock is necessary. Meantime, if we want to have only one instance for the entire JVM, using a AtomicBoolean is more appropriate.  https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/atomic/AtomicBoolean.html

What do you think? :)

Reply:
@JudeNiroshan one instance for the entire JVM is exactly what we do not want and hence the reason why we created this ticket. Did you get a chance to look at the background information on the ticket? 😊

Reply:
@dkayiwa In that case; we are good. BTW; you have already merged it  👍

Reply:
Was waiting for your feedback to revert :blush